packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts

```diff
@@ -0,0 +1,135 @@
+      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1":
+        {
+          principalId: "00000000-0000-0000-0000-000000000000",
+          clientId: "00000000-0000-0000-0000-000000000000",
+        },
+    },
+    principalId: "00000000-0000-0000-0000-000000000000",
+    tenantId: "00000000-0000-0000-0000-000000000000",
+  },
+  properties: {
+    provisioningState: "Succeeded",
+  },
+};
+
+// managed identity tracked resource
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_get = passOnSuccess([
+  mockapi.get(
+    "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+    (req) => {
+      req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+      if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+        throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+      }
+      if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+        throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+      }
+      if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+        throw new ValidationError(
+          "Unexpected managed identity resource name",
+          "identity",
+          req.params.managedIdentityResourceName,
+        );
+      }
+      return {
+        status: 200,
+        body: json(validSystemAssignedManagedIdentityResource),
+      };
+    },
+  ),
+]);
+
+Scenarios.Azure_ResourceManager_Models_CommonTypes_ManagedIdentity_ManagedIdentityTrackedResources_createWithSystemAssigned =
+  passOnSuccess([
+    mockapi.put(
+      "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Models.CommonTypes.ManagedIdentity/managedIdentityTrackedResources/:managedIdentityResourceName",
+      (req) => {
+        req.expect.containsQueryParam("api-version", "2023-12-01-preview");
+        if (req.params.subscriptionId !== SUBSCRIPTION_ID_EXPECTED) {
+          throw new ValidationError("Unexpected subscriptionId", SUBSCRIPTION_ID_EXPECTED, req.params.subscriptionId);
+        }
+        if (req.params.resourceGroup.toLowerCase() !== RESOURCE_GROUP_EXPECTED) {
+          throw new ValidationError("Unexpected resourceGroup", RESOURCE_GROUP_EXPECTED, req.params.resourceGroup);
+        }
+        if (req.params.managedIdentityResourceName.toLowerCase() !== "identity") {
+          throw new ValidationError(
+            "Unexpected managed identity resource name",
+            "identity",
+            req.params.managedIdentityResourceName,
+          );
+        }
+        req.expect.bodyEquals({
+          location: "eastus",
```

Comment:
We have some problem with empty tags, which cannot be fixed in a short time. Can we add a tag here?

Reply:
@pshao25 
Do you want to add an empty tags map in request and response data?

Reply:
Add an empty tags in `main.tsp` and `mockapi.ts`.

Reply:
I mean add a tag field with items.

Reply:
![image](https://github.com/Azure/cadl-ranch/assets/74638143/e1d0472c-9d43-4c16-8882-f023a942e81f)
Is this the tags you want to add?